addons/binding/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jpigpio.Alert;
+import jpigpio.GPIO;
+import jpigpio.JPigpio;
+import jpigpio.PigpioException;
+
+public class GPIODigitalInputHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(GPIODigitalInputHandler.class);
+    private JPigpio jPigpio = null;
+    private GPIO gpio;
+    private Date lastChanged;
+
+    public GPIODigitalInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(), getValue());
+        } catch (PigpioException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        jPigpio = ((PigpioBridgeHandler) getBridge().getHandler()).getJPiGpio();
+        try {
+            gpio = new GPIO(jPigpio, Integer.parseInt(getConfig().get(GPIO_ID).toString()), 1);
+            jPigpio.gpioSetAlertFunc(gpio.getPin(), new Alert() {
+
+                @Override
+                public void alert(int gpio, int level, long tick) {
+                    lastChanged = new Date();
+                    new Thread(new Runnable() {
+
+                        @Override
+                        public void run() {
+
+                            try {
+                                Date thisChange = new Date();
+                                OnOffType state = getValue();
+                                Thread.sleep(((BigDecimal) getConfig().get(PULLUP_TIME)).longValue());
+
+                                if ((boolean) getConfig().get(PULLUP_STRICT_MODE)) {
+                                    // Check if value changed over time
+                                    if (!thisChange.before(lastChanged)) {
+                                        updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(),
+                                                getValue());
+                                    }
+                                } else {
+                                    // Check if value is exact after time
+                                    if (state.equals(getValue())) {
+                                        updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(),
+                                                getValue());
+                                    }
+                                }
+                            } catch (InterruptedException e) {
+                                logger.error("Unknown exception", e);
```

Comment:
This is not an unknown exception this one is well defined :-)

If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/